src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs

```diff
@@ -459,6 +459,54 @@ private async Task<T> InvokeSiteExtension<T>(string subscriptionId, string resou
             return result;
         }
 
+        /// <summary>
+        /// Using this method you can invoke an API on the DaaS SiteExtension that is installed on the Web App
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="subscriptionId">Subscription Id for the resource</param>
+        /// <param name="resourceGroupName">The resource group that the resource is part of </param>
+        /// <param name="name">The resource name</param>
+        /// <param name="slotName">slot name (if querying for a slot, defaults to production slot)</param>
+        /// <param name="daasApiPath">Full path to the DAAS API </param>
+        /// <param name="apiVersion">(Optional Parameter) Pass an API version if required, 2016-08-01 is the default value</param>
+        /// <param name="cancellationToken">(Optional Parameter) Cancellation token </param>
+        /// <example>
+        /// This sample shows how to call the <see cref="InvokeDaasExtension"/> method in a detector
+        /// <code>
+        /// public async static Task<![CDATA[<Response>]]> Run(DataProviders dp, OperationContext<![CDATA[<App>]]> cxt, Response res)
+        /// {
+        ///     var resp = await dp.GeoMaster.InvokeDaasExtension<![CDATA[<dynamic>]]>(cxt.Resource.SubscriptionId, 
+        ///                     cxt.Resource.ResourceGroup, 
+        ///                     cxt.Resource.Name,
+        ///                     cxt.Resource.Slot,
+        ///                     "api/diagnosers");
+        /// 
+        ///     // do something with the response object
+        ///     var responseFromDaaS = resp.ToString();
+        ///     return res;
+        /// }
+        /// </code>
+        /// </example>
+        /// <returns></returns>
+        public async Task<T> InvokeDaasExtension<T>(string subscriptionId, string resourceGroupName, string name, string slotName, string daasApiPath, string apiVersion = GeoMasterConstants.August2016Version, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
>(string subscriptionId, string resourceGroupName, string name, string slotName, [](start = 51, length = 79)

This is a nitpik. You don't have to do this. The Diagnostics.DataProvider project is already referencing the Diagnostics.ModelsAndUtils project. Where I'm going with this is that you can create an overloaded method that just takes an IResource. IResource already has subscription, resourcegroup, and resource name information. 

This way its easier to use this method. Detector authors will just have to say `dp.GeoMaster.InvokeDaasExtension(cxt.Resource, "dasApiPath")` and thats it

Reply:
This is a really cool suggestion! let me implement this and push a commit

Reply:
I actually want to keep *both*, detectors passing cxt.Resource and all params...This would be in case if you want to do something on a slot or say a different app (I don't know maybe that scenario will arise at some point?)

Reply:
I take it back, I don't want to over publicize this method specially but this is good suggestion for any future function that I add in GeoMasterDataProvider and will keep it in mind :-)